### PR TITLE
feat(): improve translations and add es-MX

### PR DESCRIPTION
**Pull request by eduardoRoth**
This PR updates the docs project packages:
- Astro v4
- Starlight v0.15

It also includes the following:
- Translation support for all astro components. Default value when no translation is available.
- Adds es-MX language (Español México)
- Translate support for giscus
- Initial translations for es-MX

### CODE REVIEW

File: docs/src/content/docs/challenges/angular/3-directive-enhancement.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 title: 🟠 Directive Enhancement
 description: Challenge 3 is about enhancing a built-in directive
 author: thomas-laforge
-challengeNumber: 3
+challenge:
+  number: 3
```
Comment (tomalaforge) [inline docs/src/content/docs/challenges/angular/3-directive-enhancement.md:6]: why did you change `challengeNumber` into `challenge.number` ?

**Comment by tomalaforge** [comment]
Amazing work 🔥

However it's always better to divide each new features into different PR:
- upgrade of lib
- adding translation to custom element
- translating all files. 

But otherwise that's a fantastic job

Is it done  ?

File: docs/src/content/docs/challenges/angular/3-directive-enhancement.md (hunk 1 above)
Comment (eduardoRoth) [reply docs/src/content/docs/challenges/angular/3-directive-enhancement.md:6]: It was before I found how to use starlight labels object, I was sending the `challengeLabel` for the Spanish pages. When I implemented the `labels`, I left it as an object in case more information could be sent related to the challenge (maybe a custom icon, image, or something else)

**Comment by eduardoRoth** [comment]
Sorry about the massive all in one PR, I got carried away coding 😅; I still have to finish translating the challenges, but I can have that ready tomorrow. The PR can be merged and I can create another PR with the translations

File: docs/src/content/docs/challenges/angular/3-directive-enhancement.md (hunk 1 above)
Comment (tomalaforge) [reply docs/src/content/docs/challenges/angular/3-directive-enhancement.md:6]: Can you undo this change please ? We will change it if need to do it.

**Comment by tomalaforge** [comment]
No problem take your time. I'll create an issue with the mexican translation to keep track of what is done. 

Thanks again for this huge work and I'll merge it when the challenge label is undone.

**Comment by eduardoRoth** [comment]
@tomalaforge , updated.

File: docs/src/content/docs/challenges/angular/3-directive-enhancement.md (hunk 1 above)
Comment (eduardoRoth) [reply docs/src/content/docs/challenges/angular/3-directive-enhancement.md]: Updated.

**Review by tomalaforge — APPROVED** [review]

**Comment by tomalaforge** [comment]
@all-contributors add @eduardoRoth for doc
